docs/providers/aws/guide/intro.md

```diff
@@ -79,7 +79,93 @@ functions: # Your "Functions"
 resources: # The "Resources" your "Functions" use.  Raw AWS CloudFormation goes in here.
 ```
 
-When you deploy with the Framework by running `serverless deploy`, everything in `serverless.yml` is deployed at once.
+_serverless.json_
+
+```json
+{
+  "service": "users",
+  "functions": {
+    "usersCreate": {
+      "events": [
+        {
+          "http": "post users/create"
+        }
+      ]
+    },
+    "usersDelete": {
+      "events": [
+        {
+          "http": "delete users/delete"
+        }
+      ]
+    }
+  },
+  "resources": {}
+}
+```
+
+The Framework also handles natively exporting JSON object from a Javascript file `serverless.js` or from a Typescript file `serverless.ts` at the root directory of the project. While the Framework is language-agnostic, projects written in Node.js will highly benefit from using the same language for the service definition file. Such service file looks like this:
+
+_serverless.js_
+
+```js
+'use strict';
+
+module.exports = {
+  service: 'users',
+  functions: {
+    usersCreate: {
+      events: [
+        {
+          http: 'post users/create',
+        },
+      ],
+    },
+    usersDelete: {
+      events: [
+        {
+          http: 'delete users/delete',
+        },
+      ],
+    },
+  },
+  resources: {},
+};
+```
+
+_serverless.ts_
+
+```ts
+// Requiring @types/serverless in your project package.json
+import { Serverless } from 'serverless/plugins/aws/provider/awsProvider';
```

Comment:
Can you explain a bit more how resolution of that path works in context of TypeScript

First you state _"Requiring @types/serverless"_ while what's actually imported `serverless/plugins/aws/provider/awsProvider` which per Node.js rules doesn't point to any existing module.

Does TypeScript overrides Node.js native ESM support and replaces it with some different file system space?

Reply:
Hi @medikoo, `@types` is part of how typescript work and refers to definition files hosted in https://github.com/DefinitelyTyped/DefinitelyTyped . It can be confusing as it looks like you're importing from the serverless module, while all you're really doing is importing definitions from @types/serverless that will be totally erased at runtime.

You can find a discussion dividing the community in https://github.com/microsoft/TypeScript/issues/16472

Another more explicit solution is updating the import line to specify that this is solely types import using `import type` instead of `import` - https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html . The result  is the same. I updated the PR and the exemple accordingly :)

Reply:
@fredericbarthelet so in TypeScript `import` is purely for requiring from `@types`? Does it mean it's not compatible with ESM standard? e.g. you cannot use `import` to require regular Node.js ESM modules?

If so, can you please add a comment stating as:

```
// Note: In TypeScript `import` just imports type definitions from `@types/*` project (it doesn't follow standard ESM rules)
```

At least, it won't be confusing to developers which are not aware of that.

Reply:
Or probably better (more consise) comment would be:

```
// Requiring @types/serverless in your project package.json
// (In TypeScript import paths are resolved in non standard way, see: https://github.com/microsoft/TypeScript/issues/16472)
```

Reply:
> Another more explicit solution is updating the import line to specify that this is solely types import using import type instead of import... The result is the same. I updated the PR and the example accordingly :)

Thanks, that should be good enough! (sorry saw that with delay)